scripts/upload-test-results.sh

```diff
@@ -0,0 +1,16 @@
+#!/bin/bash; C:\Program\ Files\Git\bin\bash.exe
+set -euo pipefail
+IFS=$'\n\t'
+
+# upload-test-restults.sh
+#
+# SUMMARY
+#
+#   Upload `cargo-nextest` JUnit output to Datadog
+
+cd "$(dirname "${BASH_SOURCE[0]}")/.."
+set -x
+
+DD_ENV="${DD_ENV:-"local"}" datadog-ci junit upload \
+  --service vector \
+  target/nextest/default/junit.xml
```

Comment:
Would it be possible to tag these to separate out the OS / platform results? I do see in the UI that there is a section but I'm not sure where it gets that data:

<img width="257" alt="Screen Shot 2022-03-09 at 10 02 54 AM" src="https://user-images.githubusercontent.com/316880/157503142-df252a3b-07f3-493e-a60c-b86fea114c2e.png">

Reply:
Yeah, should be able to: https://docs.datadoghq.com/continuous_integration/setup_tests/junit_upload/#collecting-environment-configuration-metadata

Reply:
Added tags for arch and platform (`uname -m` and `uname -s` based)